src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Telemetry/AggregatingTelemetryLog.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.VisualStudio.Telemetry;
+using Microsoft.AspNetCore.Razor.Telemetry;
+using Microsoft.VisualStudio.Telemetry.Metrics.Events;
+using Microsoft.VisualStudio.Telemetry.Metrics;
+
+namespace Microsoft.VisualStudio.Razor.Telemetry;
+
+/// <summary>
+/// Provides a wrapper around the VSTelemetry histogram APIs to support aggregated telemetry. Each instance
+/// of this class corresponds to a specific FunctionId operation and can support aggregated values for each
+/// metric name logged.
+/// </summary>
+internal sealed class AggregatingTelemetryLog
```

Comment:
This is taken from Roslyn and adapted to how we do telemetry 

https://github.com/dotnet/roslyn/blob/main/src/VisualStudio/Core/Def/Telemetry/Shared/AggregatingTelemetryLog.cs

Reply:
I can add a comment with a permalink. The implementation itself is fairly straight forward so I wasn't sure it was needed, but I wanted to give context on where I got it for the PR.

Reply:
For sure, I just wondered if we wanted to capture the place to go look if we have questions or to see any updates.